legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalSinkCleanerAbstract.java

```diff
@@ -0,0 +1,299 @@
+        return Delete.of(metadataDataset().get(), whereCondition);
+    }
+
+    private Operation buildInsertCondition(AuditTableStatus status)
+    {
+        DatasetReference auditTableRef = this.auditDataset().get().datasetReference();
+        FieldValue tableName = FieldValue.builder().datasetRef(auditTableRef).fieldName(auditDataset().tableNameField()).build();
+        FieldValue startTs = FieldValue.builder().datasetRef(auditTableRef).fieldName(auditDataset().batchStartTimeField()).build();
+        FieldValue endTs = FieldValue.builder().datasetRef(auditTableRef).fieldName(auditDataset().batchEndTimeField()).build();
+        FieldValue batchStatus = FieldValue.builder().datasetRef(auditTableRef).fieldName(auditDataset().batchStatusField()).build();
+        FieldValue requestedBy = FieldValue.builder().datasetRef(auditTableRef).fieldName(auditDataset().requestedBy()).build();
+
+        List<org.finos.legend.engine.persistence.components.logicalplan.values.Value> fieldsToInsert = new ArrayList<>();
+        fieldsToInsert.add(tableName);
+        fieldsToInsert.add(startTs);
+        fieldsToInsert.add(endTs);
+        fieldsToInsert.add(batchStatus);
+        fieldsToInsert.add(requestedBy);
+
+        List<org.finos.legend.engine.persistence.components.logicalplan.values.Value> selectFields = new ArrayList<>();
+        selectFields.add(getMainTable());
+        selectFields.add(BatchStartTimestamp.INSTANCE);
+        selectFields.add(BatchEndTimestamp.INSTANCE);
+        selectFields.add(StringValue.of(status.name()));
+        selectFields.add(StringValue.of(requestedBy()));
+
+        return Insert.of(auditDataset().get(), Selection.builder().addAllFields(selectFields).build(), fieldsToInsert);
+    }
+
+    private StringValue getMainTable()
+    {
+        return StringValue.of(mainDataset().datasetReference().name().orElseThrow(IllegalStateException::new));
+    }
+
+    private LogicalPlan buildLogicalPlanForInitializeLock()
+    {
+        if (enableConcurrentSafety())
+        {
+            LockInfoUtils lockInfoUtils = new LockInfoUtils(lockInfoDataset().orElseThrow(IllegalStateException::new));
+            return LogicalPlan.of(Collections.singleton(lockInfoUtils.initializeLockInfo(mainDataset().datasetReference().name().orElseThrow(IllegalStateException::new), BatchStartTimestampAbstract.INSTANCE)));
+        }
+        return null;
+    }
+
+    private LogicalPlan buildLogicalPlanForAcquireLock()
+    {
+        if (enableConcurrentSafety())
+        {
+            LockInfoUtils lockInfoUtils = new LockInfoUtils(lockInfoDataset().orElseThrow(IllegalStateException::new));
+            return LogicalPlan.of(Collections.singleton(lockInfoUtils.updateLockInfo(BatchStartTimestampAbstract.INSTANCE)));
+        }
+        return null;
+    }
+
+    private SinkCleanupIngestorResult getSinkCleanupIngestorResult(SinkCleanupGeneratorResult result)
```

Comment:
We can name this one to be more explicit: executeSinkCleanup / performSinkCleanup

Reply:
+1

Reply:
done